doc/manual/src/package-management/terminology.md

```diff
@@ -0,0 +1,28 @@
+# Terminology
+
+A *local store* exists on the local filesystem of the machine where
+Nix is invoked.  The `/nix/store` directory is one example of a
+local store.  You can use other local stores by passing the
+`--store` flag to `nix`.
+
+A *remote store* is a store which exists anywhere other than the
+local filesystem.  One example is the `/nix/store` directory on
+another machine, accessed via `ssh` or served by the `nix-serve`
+Perl script.
+
+A *binary cache* is a remote store which is not the local store of
+any machine.  Examples of binary caches include S3 buckets and the
+[NixOS binary cache](https://cache.nixos.org).  Binary caches use a
+disk layout that is different from local stores; in particular, they
+keep metadata and signatures in `.narinfo` files rather than in
+`/nix/var/nix/db`.
```

Comment:
```suggestion
A *binary cache* is a specialized Nix store whose metadata and
signatures are kept in `.narinfo` files rather than in the Nix
database. Examples of binary caches include S3 buckets and the
[NixOS binary cache](https://cache.nixos.org).
```
#6265 proposes to add the term "Nix database" to the [glossary](https://nixos.org/manual/nix/unstable/glossary.html) so we should take advantage of that.

As for the differences between a Nix store and a binary cache, I couldn't find a formal specification for the former, and the latter has an in-depth treatment by @fzakaria (even [a formal spec](https://fzakaria.github.io/nix-http-binary-cache-api-spec/)) which should definitely be incorporated.